include/libaktualizr/aktualizr.h

```diff
@@ -6,12 +6,17 @@
 
 #include <boost/signals2.hpp>
 
-#include "config/config.h"
-#include "primary/events.h"
-#include "primary/secondaryinterface.h"
-#include "sotauptaneclient.h"
-#include "storage/invstorage.h"
-#include "utilities/apiqueue.h"
+#include <libaktualizr/config.h>
+#include <libaktualizr/events.h>
+
+class SecondaryInterface;
+class SotaUptaneClient;
+class INvStorage;
+class SecondaryInfo;
+
+namespace api {
+  class CommandQueue;
+}
```

Comment:
I forget, does `class api::CommandQueue` work? I thought it did and it's a little cleaner.

Reply:
yes we use it, the disadvantage he is that we have to use smart ptr, the reason is that we cant simple uses forward declaration with member objects defined by the value https://github.com/advancedtelematic/aktualizr/blob/39f58c56010f99b278267b7d9ef188c8f97e8894/include/libaktualizr/aktualizr.h#L250

Reply:
Yeah, that's fine, I just meant if we could put this forward declaration on one line by using the scope resolution operator (`::`) instead of this three-line variant.

Reply:
Why not a unique_ptr, though?

Reply:
> Yeah, that's fine, I just meant if we could put this forward declaration on one line by using the scope resolution operator (`::`) instead of this three-line variant.

Do you mean like this ```class api::CommandQueue;```

Reply:
> Why not a unique_ptr, though?

It will cause a compilation error: 
```
Invalid application of ‘sizeof’ to incomplete type ‘api::CommandQueue’
  static_assert(sizeof(_Tp)>0,
```

It can be fixed by adding an empty dtor for Aktualizr. 

Here is an answer with a good explanation
https://stackoverflow.com/a/9954553